randovania/game_description/db/resource_node.py

```diff
@@ -18,9 +19,13 @@ def is_resource_node(self) -> bool:
     def resource(self, context: NodeContext) -> ResourceInfo:
         raise NotImplementedError
 
-    def can_collect(self, context: NodeContext) -> bool:
+    def requirement_to_collect(self) -> Requirement:
```

Comment:
On a quick look, it looks like all except one implement it with `Requirement.trivial()`. So, I thought about make it the default and only let the one class override the method. But that's not really necessary IMO.

Reply:
There's likely others that could change, but I didn't try a lot. I'm doing this PR more so another PR can hopefully have no visible results.